### PR TITLE
Add mock API

### PR DESCRIPTION
**Pull request by awong-dev**
Facilitates end-to-end testing by allowing the server to be run in a mode where the test case can script API responses.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -432,7 +432,23 @@ Once selenium server is running, in a seperate terminal you should be able to ru
...
 ```
- 
+
+##### Mocking the API for e2e tests
+The backend API is a pass-through to a soap service which is not accessible during testing.
+To facilitate complete end-to-end tests, the API server can be placed into mock mode by setting
+the `HCA_MOCK_API=1` environment variable.
+
+End-to-end tests can then configure the mock by POSTing short configuration command to
+the `api/hca/v1/mock` endpoint.  Commands are are `application/json` POST bodies with the
```
Comment (akainic) [inline README.md]: Extra `are`

File: src/server/routes/mockapi.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    options.logger.info(result);
+    res.status(200).json(result);
+  });
+
+  router.all('/:resource', (req, res) => {
+    const verb = req.method.toLowerCase();
+    const verbResponses = mockResponses[verb];
+    let result = null;
+    if (verbResponses) {
+      result = verbResponses[req.params.resource];
```
Comment (akainic) [inline src/server/routes/mockapi.js]: Do you not need to return a status here?

File: README.md [hunk 3]
```diff
@@ -432,7 +432,23 @@ Once selenium server is running, in a seperate terminal you should be able to ru
 ```bash
 npm run tests:e2e
 ```
- 
+
+##### Mocking the API for e2e tests
+The backend API is a pass-through to a soap service which is not accessible during testing.
+To facilitate complete end-to-end tests, the API server can be placed into mock mode by setting
+the `HCA_MOCK_API=1` environment variable.
```
Comment (akainic) [inline README.md]: How/where do you set the environment variable?

Comment (alexose) [inline README.md]: There are a few ways to do it, but I'm always a proponent of just passing it directly to npm via the command line:

`HCA_MOCK_API=1 npm run watch`

File: src/server/routes/mockapi.js (hunk 2 above)
Comment (awong-dev) [inline src/server/routes/mockapi.js]: nope. default status is 200 which is great.

File: README.md (hunk 3 above)
Comment (awong-dev) [inline README.md]: likely we will always want to run this in nightwatch.

Comment (akainic) [inline README.md]: Can we specify that in the documentation?

**Comment by awong-dev** [comment]
merging per slack discussion

**Comment by joesoftware** [comment]
FWIW, LGTM !

**Pull request merged**